packages/validator/src/validator.ts

```diff
@@ -69,13 +86,39 @@ export class Validator {
 
     const clock = new Clock(config, logger, {genesisTime: Number(genesis.genesisTime)});
     const validatorStore = new ValidatorStore(config, slashingProtection, signers, genesis);
-    const indicesService = new IndicesService(logger, api, validatorStore);
+    this.indicesService = new IndicesService(logger, api, validatorStore);
     this.emitter = new ValidatorEventEmitter();
     this.chainHeaderTracker = new ChainHeaderTracker(logger, api, this.emitter);
     const loggerVc = getLoggerVc(logger, clock);
-    new BlockProposingService(config, loggerVc, api, clock, validatorStore, graffiti);
-    new AttestationService(loggerVc, api, clock, validatorStore, this.emitter, indicesService, this.chainHeaderTracker);
-    new SyncCommitteeService(config, loggerVc, api, clock, validatorStore, this.chainHeaderTracker, indicesService);
+
+    this.blockDutiesService = new BlockProposingService(
+      config,
+      loggerVc,
+      api,
+      clock,
+      validatorStore,
+      graffiti
+    ).dutiesService;
```

Comment:
I think it would be cleaner for each service to forward the calls, that way we don't need to expose the duties service unnecessarily. i.e. add to BlockProposingService and others
```ts
  removeDutiesForKey(pubkey: PubkeyHex): boolean {
    this.dutiesService.removeDutiesForKey(pubkey)
  }
```

Reply:
Updated as suggested